test_scripts/API/VehicleData/HandsOffSteering/common.lua

```diff
@@ -0,0 +1,369 @@
+      utils.cprint(35, "SDL was shutdown successfully")
+      isSDLShutDownSuccessfully = true
+      RAISE_EVENT(event, event)
+    end)
+  :Timeout(timeout)
+  local function forceStopSDL()
+    if isSDLShutDownSuccessfully == false then
+      utils.cprint(35, "SDL was shutdown forcibly")
+      RAISE_EVENT(event, event)
+    end
+  end
+  RUN_AFTER(forceStopSDL, timeout + 500)
+end
+
+--[[ @registerAppSuccessWithResumption: Successful application registration with custom expectations for resumption
+--! @parameters:
+--! pAppId - application number (1, 2, etc.)
+--! isHMISubscription: if false SDL resumes SubscribeVehicleData for App2 and does not send
+--! VehicleInfo.SubscribeVehicleData request to HMI
+--! @return: none
+--]]
+function m.registerAppSuccessWithResumption(pAppId, isHMISubscription)
+  if not pAppId then pAppId = 1 end
+  local handsOffSteeringResponseData = {
+    dataType = "VEHICLEDATA_HANDSOFFSTEERING",
+    resultCode = "SUCCESS"
+  }
+  m.getMobileSession(pAppId):StartService(7)
+  :Do(function()
+    local params = m.cloneTable(m.getParams(pAppId))
+    params.hashID = m.getHashId(pAppId)
+    local corId = m.getMobileSession(pAppId):SendRPC("RegisterAppInterface", params)
+    m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered", {
+      application = { appName = m.getParams(pAppId).appName }
+    })
+    m.getMobileSession(pAppId):ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+    :Do(function()
+      m.getMobileSession(pAppId):ExpectNotification("OnPermissionsChange")
+    end)
+  end)
+  if true == isHMISubscription then
+    m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData", { handsOffSteering = true })
+    :Do(function(_, data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+        { handsOffSteering = handsOffSteeringResponseData })
+    end)
+  else
+    m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData", { handsOffSteering = true }):Times(0)
+  end
+end
+
+--[[ @onVehicleData: Processing OnVehicleData notification
+--! @parameters:
+--! pHandsOffSteering: Vehicle parameter value
+--! pExpTimes:  expected number of notifications
+--! pAppId: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.onVehicleData(pHandsOffSteering, pExpTimes, pAppId)
+  if not pExpTimes then pExpTimes = 1 end
+  if not pAppId then pAppId = 1 end
+  m.getHMIConnection():SendNotification("VehicleInfo.OnVehicleData", { handsOffSteering = pHandsOffSteering })
+  m.getMobileSession(pAppId):ExpectNotification("OnVehicleData", { handsOffSteering = pHandsOffSteering })
+  :Times(pExpTimes)
+end
+
+--[[ @onVehicleDataForTwoApps: Processing OnVehicleData notification for two apps
+--! @parameters:
+--! pHandsOffSteering: Vehicle parameter value
+--! pAppId_1, pAppId_2: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.onVehicleDataForTwoApps(pHandsOffSteering, pAppId_1, pAppId_2)
```

Comment:
We know that function is used for 2 apps, so appId can not be used as params and use them directly in the function